nexosim/src/server/services/inspector_service.rs

```diff
@@ -0,0 +1,583 @@
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+
+use crate::registry::{EventSinkRegistry, EventSourceRegistry, QuerySourceRegistry};
+
+use super::super::codegen::simulation::*;
+use super::{map_registry_error, simulation_not_started_error};
+
+/// Protobuf-based simulation inspector.
+///
+/// An `InspectorService` provides information about available endpoints and
+/// types of the messages passed.
+pub(crate) enum InspectorService {
+    NotStarted,
+    Started {
+        event_sink_registry: Arc<Mutex<EventSinkRegistry>>,
+        event_source_registry: Arc<EventSourceRegistry>,
+        query_source_registry: Arc<QuerySourceRegistry>,
+    },
+}
+
+impl InspectorService {
+    /// Returns a list of names of all the registered event sources.
+    pub(crate) fn list_event_sources(&self, _: ListEventSourcesRequest) -> ListEventSourcesReply {
+        match self {
+            Self::Started {
+                event_source_registry,
+                ..
+            } => ListEventSourcesReply {
+                source_names: event_source_registry
+                    .list_sources()
+                    .map(|a| a.to_string())
+                    .collect(),
+                result: Some(list_event_sources_reply::Result::Empty(())),
+            },
+            Self::NotStarted => ListEventSourcesReply {
+                source_names: Vec::new(),
+                result: Some(list_event_sources_reply::Result::Error(
+                    simulation_not_started_error(),
+                )),
+            },
+        }
+    }
+
+    /// Retrieves the input event schemas for the specified event sources.
+    ///
+    /// If the `source_names` field is left empty, it returns schemas for all
+    /// the registered sources.
```

Comment:
Maybe we should document as well that "raw" events will return an empty string, rather than not being listed at all.